src/rp2_common/pico_double/include/pico/double.h

```diff
@@ -16,64 +16,195 @@ extern "C" {
-* - powint, sincos (GNU extensions)
+* 1. To use optimized software implementations provided by the RP2-series device's bootrom or the SDK
+* 2. To use optimized combined software/hardware implementations utilizing custom RP2-series hardware for acceleration
+* 3. To control the amount of C compiler/library code bloat
+* 4. To make sure no floating point is called at all
 *
-* The following additional optimized functions are also provided:
+* The pico_double library comes in three main flavors:
 *
-* - int2double, uint2double, int642double, uint642double, fix2double, ufix2double, fix642double, ufix642double
-* - double2fix, double2ufix, double2fix64, double2ufix64, double2int, double2uint, double2int64, double2uint64, double2int_z, double2int64_z,
-* - exp10, sincos, powint
+* 1. `pico_double_none` - all floating point operations cause a \ref panic - no double-precision floating point code is included
+* 2. `pico_double_compiler` - no custom functions are provided; all double-precision floating point is handled by the C compiler/library
+* 3. `pico_double_pico` - the smallest and fastest available for the platform, along with additional functionality (e.g. fixed point conversions) which are detailed below
 *
-* On RP2350 the following additional functions are available; the _fast methods are faster but do not round correctly"
+* The user can control which version they want (e.g. **pico_double_xxx** by either setting the CMake global variable
+* `PICO_DEFAULT_DOUBLE_IMPL=xxx`, or by using the CMake function `pico_set_double_implementation(<TARGET> xxx)`. Note that in the absence
+* of either, pico_double_pico is used by default.
 *
-* - ddiv_fast, sqrt_fast
+* \if rp2040_specific
+* On RP2040, `pico_double_pico` uses optimized hand coded implementations from the bootrom and the SDK for both
+* basic double-precision floating point operations and floating point math library functions. These implementations
+* are generally faster and smaller than those provided by the C compiler/library, though they don't support all the features of a fully compliant
+* floating point implementation; they are however usually fine for the majority of cases
+* \endif
+*
+* \if rp2350_specific
+* On RP2350, `pico_double_pico` uses RP2350 DCP instructions (double co-processor) to implement fast version of the basic
+* arithmetic functions, and provides optimized M33 implementations of trignometric and scientific functions.
+* These implementations are generally faster and smaller than those provided by the C compiler/library, though they don't support all the features of a fully compliant
+* floating point implementation; they are however usually fine for the majority of cases
+* \endif
+*
+* On Arm, (replacement) optimized implementations are provided for the following compiler built-ins when
+* and math library functions when using `pico_dobule_pico`:
```

Comment:
Some kind of copy-paste typo? "compiler built-ins when and math library functions"